cfy_cluster_manager/main.py

```diff
@@ -195,51 +205,162 @@ def _prepare_config_files(instances_dict,
 
 
 def _install_cloudify_remotely(instance, rpm_download_link):
+    logger.info('Downloading Cloudify RPM on %s from %s',
+                instance.name, rpm_download_link)
+    instance.run_command('curl -o {0} {1}'.format(
+        RPM_PATH, rpm_download_link), hide_stdout=True)
+    logger.info('Installing Cloudify RPM on %s', instance.name)
+    instance.run_command(
+        'yum install -y {}'.format(RPM_PATH), use_sudo=True, hide_stdout=True)
+
+
+def _get_service_status_code(instance):
+    """Checking the status code of the cfy_manager_install_<type> service."""
+    result = instance.run_command(
+        'systemctl status {}'.format(instance.unit_name),
+        use_sudo=True, hide_stdout=True, ignore_failure=True)
+    return result.return_code
+
+
+def _wait_for_cloudify_current_installation(instance):
+    logger.info(
+        'Waiting for current installation of %s to finish', instance.name)
+    status_code = 0
+    timeout = 0
+    while status_code == 0 and timeout <= 600:
+        if timeout % 20 == 0:
+            logger.info('Waiting for current installation of %s to finish',
+                        instance.name)
+        time.sleep(1)
+        timeout += 1
```

Comment:
This isn't a brilliantly named variable- I'd normally expect timeout to remain constant and current_time or run_time to increase.
Alternatively I'd consider doing it with attempts vs max_attempts.

Reply:
Changed to retry_count (i.e. max_attempts).